File: app/src/mock/java/torille/fi/lurkforreddit/data/remote/FakeRedditRepository.java
package torille.fi.lurkforreddit.data.remote;

import android.support.annotation.NonNull;
import android.support.v4.util.Pair;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Observable;
import torille.fi.lurkforreddit.data.RedditDataSource;
import torille.fi.lurkforreddit.data.RedditRepository;
import torille.fi.lurkforreddit.data.models.view.Comment;
import torille.fi.lurkforreddit.data.models.view.Post;
import torille.fi.lurkforreddit.data.models.view.SearchResult;
import torille.fi.lurkforreddit.data.models.view.Subreddit;

/**
 * Created by eva on 25.4.2017.
 */

public class FakeRedditRepository implements RedditDataSource {

    @Inject
    public FakeRedditRepository() {
    }

    @Override
    public Observable<List<Subreddit>> getSubreddits() {
        return null;
    }

    @Override
    public Observable<Pair<String, List<Post>>> getSubredditPosts(@NonNull String subredditUrl) {
        return null;
    }

    @Override
    public Observable<Pair<String, List<Post>>> getMoreSubredditPosts(@NonNull String subredditUrl, @NonNull String nextpageId) {
        return null;
    }

    @Override
    public void refreshData() {

    }

    @Override
    public Observable<List<Comment>> getCommentsForPost(@NonNull String permaLinkUrl) {
        return null;
    }

    @Override
    public Observable<List<Comment>> getMoreCommentsForPostAt(@NonNull List<String> childCommentIds, @NonNull String linkId, int commentLevel) {
        return null;
    }

    @Override
    public Observable<Pair<String, List<SearchResult>>> getSearchResults(@NonNull String query) {
        return null;
    }

    @Override
    public Observable<Pair<String, List<SearchResult>>> getMoreSearchResults(@NonNull String query, @NonNull String after) {
        return null;
    }
}

File: app/src/mock/java/torille/fi/lurkforreddit/data/remote/FakeRedditRepository.kt
package torille.fi.lurkforreddit.data.remote

import android.support.v4.util.Pair
import io.reactivex.Observable
import torille.fi.lurkforreddit.data.RedditDataSource
import torille.fi.lurkforreddit.data.models.view.*
import torille.fi.lurkforreddit.di.scope.RedditScope
import javax.inject.Inject

/**
 * Created by eva on 25.4.2017.
 */
@RedditScope
class FakeRedditRepository @Inject
internal constructor() : RedditDataSource {

    override fun getCommentsForPost(permaLinkUrl: String, isSingleCommentThread: Boolean): Observable<PostAndComments> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getSubreddits(): Observable<List<Subreddit>> {
        return Observable.fromArray(emptyList())
    }

    override fun getSubredditPosts(subredditUrl: String): Observable<Pair<String, List<Post>>> {
        return Observable.fromArray(Pair("first", emptyList()))
    }

    override fun getMoreSubredditPosts(subredditUrl: String, nextpageId: String): Observable<Pair<String, List<Post>>> {
        return Observable.fromArray(Pair("first", emptyList()))
    }

    override fun refreshData() {

    }

    fun getCommentsForPost(permaLinkUrl: String): Observable<List<Comment>>? {
        return null
    }

    override fun getMoreCommentsForPostAt(childCommentIds: List<String>, linkId: String, commentLevel: Int): Observable<List<Comment>> {
        return Observable.fromArray(emptyList())
    }

    override fun getSearchResults(query: String): Observable<Pair<String, List<SearchResult>>> {
        return Observable.fromArray(Pair("first", emptyList()))
    }

    override fun getMoreSearchResults(query: String, after: String): Observable<Pair<String, List<SearchResult>>> {
        return Observable.fromArray(Pair("first", emptyList()))
    }
}
